pymc3_models/models/NaiveBayes.py

```diff
@@ -0,0 +1,283 @@
+        categorical distribution:
+
+        .. math::
+
+            P(y=c|x_i) = Cat(\\pi_1, \dots, \\pi_C)
+
+        where :math:`\pi_i` is the probability that a vector belongs to category :math:`i`.
+
+        We assume that the :math:`\pi_i` follow a Dirichlet distribution:
+
+        .. math::
+
+            \\pi \\sim Dirichlet(\\alpha)
+
+        with hyperparameter :math:`\\alpha = [1, .., 1]`. The :math:`\\mu_{jc}`
+        are sampled from a Normal distribution centred on :math:`0` with
+        variance :math:`100`, and the :math:`\\sigma_{jc}` are sampled from a
+        HalfNormal distribuion of variance :math:`100`:
+
+        .. math::
+
+            \\mu_{jc} \\sim Normal(0, 100)
+
+            \\sigma_{jc} \\sim HalfNormal(100)
+
+        Note that the Gaussian Naive Bayes model is equivalent to a Gaussian
+        mixture with a diagonal covariance [1].
+
+        Returns
+        -------
+        A PyMC3 model
+
+        References
+        ----------
+        .. [1] Murphy, K. P. (2012). Machine learning: a probabilistic perspective.
+        """
+
+        # The data
+        X = theano.shared(np.zeros((self.num_training_samples, self.num_pred)))
+        y = theano.shared(np.zeros(self.num_training_samples))
+
+        self.shared_vars = {
+            'model_input': X,
+            'model_output': y
+        }
+
+        model = pm.Model()
+        with model:
+            # Priors
+            alpha = np.ones(self.num_cats)
+            pi = pm.Dirichlet('pi', alpha, shape=self.num_cats)
+            mu = pm.Normal('mu', 0, 100, shape=(self.num_cats, self.num_pred))
+            sigma = pm.HalfNormal('sigma', 100, shape=(self.num_cats, self.num_pred))
+
+            # Assign classes to data points
+            z = pm.Categorical('z', pi, shape=self.num_training_samples, observed=y)
+
+            # The components are independant and normally distributed
+            xi = pm.Normal('xi', mu[z], sigma[z], observed=X)
+
+        return model
+
+    def fit(self, X, y, inference_type='advi', minibatch_size=None, inference_args=None):
+        """
+        Train the Naive Bayes model.
+
+        Parameters
+        ----------
+        X : numpy array, shape [num_training_samples, num_pred]. Contains the data points.
+
+        y : numpy array, shape [num_training_samples,]. Contains the category of the data points.
+
+        inference_type : string, specifies which inference method to call.
+            Default is 'advi'. Currently, only 'advi' and 'nuts'are implemented.
+
+        minibatch_size : int, number of samples to include in each minibatch
+            for ADVI. Defaults to None so minibatch is not run by default.
+
+        inference_args : dict, arguments to be passed to the inference methods.
+            Check the PyMC3 documentation.
+
+        Returns
+        -------
+        The current instance of the PyMC3 model.
```

Comment:
This is potentially confusing because it returns the class instance, not the PyMC3 model instance. Could you clarify this to make it clear it's an instance of GaussianNaiveBayes and not pm.Model?